Add blog post on the v5 gem releases

peteryates opened the pull request.

paulrobertlloyd commented on app/posts/releasing-rails-libraries-for-version-5.md:
```diff
@@ -0,0 +1,120 @@
+  url: https://github.com/peteryates
+---
+
+Version 5 of the GOV.UK Design System was released on December 8th 2023. Half an hour later, we released version 5 of the x-govuk [form builder](https://govuk-form-builder.netlify.app/) and [components](https://govuk-components.netlify.app/) libraries.
```
Use GOV.UK style for dates:

```suggestion
Version 5 of the GOV.UK Design System was released on 8 December 2023. Half an hour later, we released version 5 of the x-govuk [form builder](https://govuk-form-builder.netlify.app/) and [components](https://govuk-components.netlify.app/) libraries.
```

paulrobertlloyd commented on app/posts/releasing-rails-libraries-for-version-5.md:
```diff
@@ -0,0 +1,120 @@
+
+Version 5 of the GOV.UK Design System was released on December 8th 2023. Half an hour later, we released version 5 of the x-govuk [form builder](https://govuk-form-builder.netlify.app/) and [components](https://govuk-components.netlify.app/) libraries.
+
+The short turnaround time might suggest it was a quick job, but work on the included features started in May and we've made some significant improvements since then.
```
Smart quote for smart people. (There are likely others that need updating)

```suggestion
The short turnaround time might suggest it was a quick job, but work on the included features started in May and we’ve made some significant improvements since then.
```

peteryates replied:
Why update them when the build progress automatically fixes them (thanks `smartypants`!)

paulrobertlloyd replied:
It’s a good question, and touches on an existential crisis. Perhaps we leave these and stop trying to catch issues in PRs (and I can update previous posts to use ASCII to satisfy my OCD needs)

frankieroberto replied:
@paulrobertlloyd that really is OCD! Maybe we should follow the principle of being lax about input (curly or straight quotes) and strict about output...? 😄

paulrobertlloyd replied:
I’ve really got a lot of issues…

![Scene from ‘I’m Alan Partidge’ where he responds with: “yeah, of What Car? Magazine“."](https://img.gifglobe.com/grabs/partridgecloud/S02E06/gif/vPCVHa3Qnqqr.gif)

paulrobertlloyd commented on app/posts/releasing-rails-libraries-for-version-5.md:
```diff
@@ -0,0 +1,120 @@
+
+> Time spent on [boilerplate](https://en.wikipedia.org/wiki/Boilerplate_code) is time spent not focussing on user needs.
+
+When working on anything that's used by lots of teams, like [referring serious teacher misconduct](https://refer-serious-misconduct.education.gov.uk), [whistleblowing at a company](https://make-a-business-whistleblower-report.service.gov.uk/) or even [GOV.UK forms itself](https://www.forms.service.gov.uk/), there's a great deal of pressure not to break anything. This responsibility forces us to m ...
```
Product name is capitalised:

```suggestion
When working on anything that's used by lots of teams, like [referring serious teacher misconduct](https://refer-serious-misconduct.education.gov.uk), [whistleblowing at a company](https://make-a-business-whistleblower-report.service.gov.uk/) or even [GOV.UK Forms itself](https://www.forms.service.gov.uk/), there's a great deal of pressure not to break anything. This responsibility forces us to make changes with a degree of caution and to rigorously test and review.
```

paulrobertlloyd commented on app/posts/releasing-rails-libraries-for-version-5.md:
```diff
@@ -0,0 +1,120 @@
+
+There are two main reasons for us making changes:
+
+* **following upstream changes** --- like adding a new GOV.UK component or adding support for a new version of Ruby or Rails
+* **improvements we feel make the library better** --- like adding features, tidying code or making the tests run faster
```
We can use Unicode characters (although I’m constantly in debate with myself about how much Markdown should only use ASCII characters or not)

```suggestion
* **following upstream changes** — like adding a new GOV.UK component or adding support for a new version of Ruby or Rails
* **improvements we feel make the library better** — like adding features, tidying code or making the tests run faster
```

paulrobertlloyd replied:
Prettier will likely flag:

```suggestion
- **following upstream changes** --- like adding a new GOV.UK component or adding support for a new version of Ruby or Rails
- **improvements we feel make the library better** --- like adding features, tidying code or making the tests run faster
```

paulrobertlloyd commented on app/posts/releasing-rails-libraries-for-version-5.md:
```diff
@@ -0,0 +1,120 @@
+
+## The task list
+
+The [GOV.UK design system task list component](https://design-system.service.gov.uk/components/task-list/) has been a long time in the making with discussions [stretching back to 2018](https://github.com/alphagov/govuk-design-system-backlog/issues/72). However, thanks to Frankie Roberto and the GOV.UK design system team's tireless efforts, [the PR](https://github.com/alphagov/govuk-frontend/pull/ ...
```
Product name is capitalised:

```suggestion
The [GOV.UK Design System task list component](https://design-system.service.gov.uk/components/task-list/) has been a long time in the making with discussions [stretching back to 2018](https://github.com/alphagov/govuk-design-system-backlog/issues/72). However, thanks to Frankie Roberto and the GOV.UK Design System team's tireless efforts, [the PR](https://github.com/alphagov/govuk-frontend/pull/2261) was finally approved and merged in June.
```

paulrobertlloyd replied:
Avoid abbreviations where possible:

```suggestion
The [GOV.UK design system task list component](https://design-system.service.gov.uk/components/task-list/) has been a long time in the making with discussions [stretching back to 2018](https://github.com/alphagov/govuk-design-system-backlog/issues/72). However, thanks to Frankie Roberto and the GOV.UK design system team's tireless efforts, [the pull request](https://github.com/alphagov/govuk-frontend/pull/2261) was finally approved and merged in June.
```

paulrobertlloyd commented on app/posts/releasing-rails-libraries-for-version-5.md:
```diff
@@ -0,0 +1,120 @@
+* might not be able to do them all in one sitting
+* is able to complete them in any order
+
+Like with the other components, the Rails implementation borrows heavily from the GOV.UK design system reference, but translates it into an syntax that will be immediately familiar to Ruby on Rails developers.
```
```suggestion
Like with the other components, the Rails implementation borrows heavily from the GOV.UK Design System reference, but translates it into an syntax that will be immediately familiar to Ruby on Rails developers.
```

paulrobertlloyd commented on app/posts/releasing-rails-libraries-for-version-5.md:
````diff
@@ -0,0 +1,120 @@
+<a class="govuk-link govuk-link--no-visited-state" target="_blank" rel="noreferrer noopener">
+  View<span class="govuk-visually-hidden"> my profile</span>
+</a>
+```
````
Is there a concluding paragraph we can add to this post, as it seems to end abruptly. Maybe link to the documentation site and/or the repo to provide feedback?

frankieroberto replied:
Agree, it does end slightly abruptly. Maybe a call for feedback and/or a suggestion that the project will continue to both iterate the syntax and follow upstream changes to the design system?

paulrobertlloyd reviewed (changes requested):
Apologies, some of my feedback is likely quite nit-picky, which doesn’t accurately reflect my joy that you’ve written this post!

paulrobertlloyd commented on app/posts/releasing-rails-libraries-for-version-5.md:
```diff
@@ -0,0 +1,120 @@
+
+The above snippet will render this task list:
+
+![A GOV.UK design system task list with entries for personal and contact details, and a greyed out identity check task that cannot yet be started](/assets/posts/releasing-rails-libraries-for-version-5/rendered-task-list.png)
```
Image is missing from this pull request.

frankieroberto commented on app/posts/releasing-rails-libraries-for-version-5.md:
```diff
@@ -0,0 +1,123 @@
+
+The short turnaround time might suggest it was a quick job, but work on the included features started in May and we've made some significant improvements since then.
+
+For the uninitiated, the form builder and components libraries are implementations of the [GOV.UK Design System](https://design-system.service.gov.uk/) for [Ruby on Rails](https://rubyonrails.org/), one of the most popular web frameworks used across government.
```
```suggestion
The form builder and components libraries are implementations of the [GOV.UK Design System](https://design-system.service.gov.uk/) for [Ruby on Rails](https://rubyonrails.org/), one of the most popular web frameworks used across government.
```

frankieroberto commented on app/posts/releasing-rails-libraries-for-version-5.md:
```diff
@@ -0,0 +1,123 @@
+
+In November 2022 [I tried to add a new feature](https://github.com/x-govuk/govuk-components/pull/363) to `govuk_link_to` which allowed developers to call it with `new_tab: true`, and the resulting link would open pages in a new tab.
+
+Unfortunately [I had to revert the change shortly afterwards](https://github.com/x-govuk/govuk-components/pull/399) because mixing new positional keyword arguments, which were added to Ruby in 2019 with Rails' link helpers, which haven't really changed since 2004, caused lots of problems. Thankfully we uncovered these in our pre-release testing so didn't break anything important.
```
I found this paragraph slightly confusing - do you mean "mixing keyword arguments _with_ positional arguments"?

peteryates replied:
Yeah this one's a nightmare, will redo

frankieroberto reviewed (commented):
This is a great! Left a few suggestions for minor tweaks, and agree some kind of short concluding paragraph would be good.

peteryates commented:
Totally agree on the concluding paragraph. An earlier draft had a 'What's next' section which I think I'll re-add.

peteryates commented:
Thanks @paulrobertlloyd, @frankieroberto - think I've addressed everything and cleaned up some of the waffle 😅 🧇

Please feel free to have anther peruse through and hopefully we can get this one out tomorrow.

frankieroberto commented on app/posts/releasing-rails-libraries-for-version-5.md:
```diff
@@ -0,0 +1,131 @@
+
+## What's next
+
+Now version 5 is released we'll slow the pace for a while and let the dust settle. This should give developers the opportunity to try out the new features and provide feedback or report bugs.
```
Make this slightly more direct? Should we say how to report bugs/give feedback (ie link to GitHub issues?)

```suggestion
Now version 5 is released, we'll slow the pace for a while and let the dust settle to give you the opportunity to try out the new features.

Please report any bugs, and give us your feedback.
```

paulrobertlloyd commented:
Lovely stuff. When you’re ready, squash and merge, and then share your post with wild abandon! 🥳
